### PR TITLE
feat(sessionclaims): support for email verification claim

### PR DESCRIPTION
**Pull request by porcellus**
## Summary of change

Add email verification claim

## Related issues

-  https://github.com/supertokens/supertokens-node/pull/349
-  https://github.com/supertokens/supertokens-web-js/pull/48

## Test Plan

Done separately

## Documentation changes

Done separately

## Checklist for important updates
- [ ] Changelog has been updated
- [x] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
   - Along with the associated array in `lib/ts/version.ts`
- [x] Changes to the version if needed
   - In `package.json`
   - In `package-lock.json`
   - In `lib/ts/version.ts`
- [x] Had run `npm run build-pretty`
- [x] Had installed and ran the pre-commit hook
- [x] Issue this PR against the latest non released version branch.
   - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
   - If no such branch exists, then create one from the latest released branch.

### CODE REVIEW

File: lib/ts/recipeImplementation.ts [hunk 1]
```diff
@@ -199,7 +199,8 @@ export default function RecipeImplementation(recipeImplInput: {
                 if (!validationRes.isValid) {
                     errors.push({
                         validatorId: validator.id,
-                        reason: validationRes.reason
+                        reason: validationRes.reason,
+                        redirectPath: validationRes.redirectPath
```
Comment (rishabhpoddar) [inline lib/ts/recipeImplementation.ts:203]: Im not sure I quite understand this - where does `redirectPath` come from? If it comes from the backend, how would the backend know what the path is?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: lib/ts/recipeImplementation.ts (hunk 1 above)
Comment (rishabhpoddar) [reply lib/ts/recipeImplementation.ts:203]: Please figure out a way to remove redirectToPath from website and web-js and adding that specific field in auth-react. Some ideas we discussed:
- Adding a callback to the auth-react's `validateGlobalClaimsAndRedirect` which will be called when claim validations fail.
- Making the error type generic here.

File: lib/ts/recipeImplementation.ts [hunk 2]
```diff
@@ -199,7 +199,8 @@ export default function RecipeImplementation(recipeImplInput: {
...
                     });
                 }
```
Comment (rishabhpoddar) [inline lib/ts/recipeImplementation.ts:205]: We should add a test to enforce that the claims sent to `validateClaims` array are indeed checked in the same order as they are in the array. Which means that the error[] returned should have the errors in the same order as well.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: lib/ts/recipeImplementation.ts (hunk 2 above)
Comment (porcellus) [reply lib/ts/recipeImplementation.ts:205]: done in #143

**Comment by porcellus** [comment]
Decided to go in a different direction that doesn't need an interface change here.

**Pull request closed without merging**